### PR TITLE
Toast

### PR DESCRIPTION
**Pull request by dominikwilkowski**
The plan is:

- [x] separated window_metrics out of tooltips
- [x] create a new primitive for toasts that have access to window size
- [x] find a way to position the toast outside of scrolls so we don't have to worry about those
- [x] auto dismiss them
- [ ] ~stop auto dismissing when hovered~
- [ ] ~manual dismiss button~

### CODE REVIEW

File: src/ui/app_view.rs [hunk 1]
```diff
@@ -57,12 +60,13 @@ pub fn app_view(
...
 
-	let tooltip_signals_settings = TooltipSignals::new(que);
+	let que_settings = Que::default();
+	let tooltip_signals_settings = TooltipSignals::new(que_settings);
 	let overflow_labels = create_rw_signal(vec![0]);
 
 	let field_presets = create_rw_signal(env.config.get_field_presets());
 
-	let clear_icon = include_str!("./icons/clear.svg");
+	let delete_icon = include_str!("./icons/delete.svg");
```
Comment (dominikwilkowski) [inline src/ui/app_view.rs:69]: We had two icons that looked the exact same... removed the clear icon and just re-used the one we had

File: src/main.rs [hunk 2]
```diff
@@ -162,14 +152,16 @@ fn main() {
...
+							app_state.set(AppState::Ready);
+						});
 					},
 					Err(err) => {
-						eprintln!("{:#?}", err);
-						error.set(err.to_string());
-						app_state.set(AppState::Ready);
+						untrack(|| {
+							toast_signals.add(err.to_string());
+						});
```
Comment (dominikwilkowski) [inline src/main.rs:164]: This took me ages to figure out. Without `untrack` this lands us in an infinite loop. So I also added it to the `Ok` branch as I felt the signals feel a bit brittle. Now we're explicit that we don't want to run the effect again on changes inside of it.

File: src/ui/primitives/tooltip.rs [hunk 3]
```diff
@@ -38,10 +38,6 @@ impl TooltipSignals {
 		self.que.tooltip.update(|item| item.retain(|ids| *ids != id));
 	}
 
-	pub fn unque_all_tooltips(self) {
-		self.que.tooltip.set(Vec::new());
-	}
-
```
Comment (dominikwilkowski) [inline src/ui/primitives/tooltip.rs:44]: Felt like this was not the right place for this so I moved it to the Que instance

File: src/ui/settings/editing.rs [hunk 4]
```diff
@@ -65,7 +65,7 @@ fn delete_preset(
 	field_presets.set(presets);
 }
 
-fn prefix_line(
+fn preset_line(
```
Comment (dominikwilkowski) [inline src/ui/settings/editing.rs:68]: No idea why I named this `prefix` instead of `preset`... don't drink and code kids!

**Review by thewizzy — APPROVED** [review]
Thank you for the comments, that was a huge PR but the bits refactored made sense.

**Pull request merged**